src/web3/KeychainSigner.ts

```diff
@@ -229,37 +59,8 @@ export class KeychainSigner implements Signer {
     if (!privateKey) {
       return false
     }
-
-    this.unlockedLocalSigner = new LocalSigner(privateKey)
-    this.unlockTime = Date.now()
-    this.unlockDuration = duration
-    return true
-  }
-
-  isUnlocked(): boolean {
-    if (this.unlockDuration === undefined || this.unlockTime === undefined) {
-      return false
-    }
-
-    if (this.unlockDuration === 0) {
-      return true
-    }
-
-    return this.unlockTime + this.unlockDuration * 1000 > Date.now()
-  }
-
-  /**
-   * Updates the passphrase of an account
-   * @param oldPassphrase - the passphrase currently associated with the account
-   * @param newPassphrase - the new passphrase to use with the account
-   * @returns whether the update was successful
-   */
-  async updatePassphrase(oldPassphrase: string, newPassphrase: string) {
-    const privateKey = await getStoredPrivateKey(this.account, oldPassphrase)
-    if (!privateKey) {
-      return false
-    }
-    await storePrivateKey(privateKey, this.account, newPassphrase)
+    this.signer = new LocalSigner(privateKey)
```

Comment:
still has the functionality of the new signer getting created on every unlock (not sure if thats actually needed)